.github/workflows/unit-tests.yml

```diff
@@ -0,0 +1,97 @@
+name: Unit tests
+
+on:
+  pull_request:
+  push:
+
+jobs:
+  style:
+    name: Style
+    runs-on: ubuntu-latest
+    steps:
+      - name: Clone project
+        uses: actions/checkout@v2
+      - name: hlint
+        run: |
+          set -ex
+          ./etc/scripts/get-hlint.sh
+          export PATH="$(pwd)"/hlint:$PATH
+          hlint src/
+          hlint src/ --cpp-define=WINDOWS=1
+          hlint test/ --cpp-simple
+  pedantic:
+    name: Pedantic
+    runs-on: ubuntu-latest
+    steps:
+      - name: Clone project
+        uses: actions/checkout@v2
+      - name: Cache dependencies
+        uses: actions/cache@v1
+        with:
+          path: ~/.stack
+          key: ${{ runner.os }}-${{ hashFiles('stack.yaml') }}
+          restore-keys: |
+            ${{ runner.os }}-
+      - name: Pedantic build
+        run: stack --system-ghc build --pedantic
+  unit-tests:
+    name: Unit tests
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+        - os: ubuntu-latest
+          stack-yaml: stack-ghc-84.yaml
+          extra-suffix: ""
+          stack-args: ""
+        - os: ubuntu-latest
+          stack-yaml: stack-ghc-86.yaml
+          extra-suffix: ""
+          stack-args: ""
+        - os: ubuntu-latest
+          stack-yaml: stack-ghc-88.yaml
+          extra-suffix: ""
+          stack-args: ""
+        - os: ubuntu-latest
+          stack-yaml: stack.yaml
+          extra-suffix: "alpine"
+          stack-args: "--docker --system-ghc --no-install-ghc --flag stack:static"
+        - os: windows-latest
+          stack-yaml: stack.yaml
+          extra-suffix: ""
+          stack-args: ""
+        - os: macos-latest
+          stack-yaml: stack.yaml
+          extra-suffix: ""
+          stack-args: ""
+    steps:
+      - name: Clone project
+        uses: actions/checkout@v2
+      - name: Cache dependencies
+        uses: actions/cache@v1
+        with:
+          path: ~/.stack
+          key: ${{ runner.os }}-${{ hashFiles('${{ matrix.stack-yaml }}') }}-${{ matrix.extra-suffix }}
```

Comment:
Do you think `package.yaml` should be part of the cache hash ?

Reply:
I could go both ways, but I lean towards no. Basically: if package.yaml changes, but stack.yaml doesn't, it means that packages were added or removed, but the package set itself is unchanged. That means that the cache shouldn't be invalidated.